test/integration/scheduler/queue_test.go

```diff
@@ -148,3 +161,174 @@ func TestServiceAffinityEnqueue(t *testing.T) {
 		t.Errorf("Expected the Pod to be attempted 2 times, but got %v", podInfo.Attempts)
 	}
 }
+
+var _ framework.FilterPlugin = &fakeCRPlugin{}
+var _ framework.EnqueueExtensions = &fakeCRPlugin{}
+
+type fakeCRPlugin struct{}
+
+func (f *fakeCRPlugin) Name() string {
+	return "fakeCRPlugin"
+}
+
+func (f *fakeCRPlugin) Filter(_ context.Context, _ *framework.CycleState, _ *v1.Pod, _ *framework.NodeInfo) *framework.Status {
+	return framework.NewStatus(framework.Unschedulable, "always fail")
+}
+
+// EventsToRegister returns the possible events that may make a Pod
+// failed by this plugin schedulable.
+func (f *fakeCRPlugin) EventsToRegister() []framework.ClusterEvent {
+	return []framework.ClusterEvent{
+		{Resource: "foos.v1.example.com", ActionType: framework.All},
+	}
+}
+
+// TestCustomResourceEnqueue constructs a fake plugin that registers custom resources
+// to verify Pods failed by this plugin can be moved properly upon CR events.
+func TestCustomResourceEnqueue(t *testing.T) {
+	// Start API Server with apiextensions supported.
+	server := apiservertesting.StartTestServerOrDie(
+		t, apiservertesting.NewDefaultTestServerOptions(),
+		[]string{"--disable-admission-plugins=ServiceAccount,TaintNodesByCondition", "--runtime-config=api/all=true"},
+		testfwk.SharedEtcd(),
+	)
+	testCtx := &testutils.TestContext{}
```

Comment:
importing an integration test pkg here is strange, why not have this test as an integration test?

Reply:
I don't follow that. It's an integration test.

Reply:
ah, sorry, I didn't notice, I thought it was a unit test :)